lib/plugins/rotate_asg_instances/ssh.rb

```diff
@@ -0,0 +1,44 @@
+module Moonshot
+  module Asg
+    class SSHValidationError < StandardError; end
+
+    class SSH
+      def initialize(resources, instance_id = first_instance_id(resources))
+        @config = resources.controller.config
+        @config.ssh_instance = instance_id
+      end
+
+      def try_to_ssh
+        test_ssh_connection
+      rescue SSHValidationError => e
+        raise e.class,
+              "SSH failed. exit status: #{@command_response.exitstatus} " \
+              "output: #{@command_response.output} " \
+              "error: #{@command_response.error}"
+      end
+
+      def exec(command)
+        cb = SSHCommandBuilder.new(@config.ssh_config, @config.ssh_instance)
+        SSHForkExecutor.new.run(cb.build(command).cmd)
+      end
+
+      private
+
+      def test_ssh_connection
+        Retriable.retriable(base_interval: 5, tries: 3) do
+          @command_response = exec('/bin/true')
+          raise SSHValidationError, 'Validation failed.' unless
+            @command_response.exitstatus.zero?
+          @command_response.exitstatus.zero?
+        end
+      end
+
+      def first_instance_id(resources)
```

Comment:
I think this method belongs to asg class.

Reply:
Moved back